apps/server/src/modules/learnroom/utils/common-cartridge-file-validator.pipe.ts

```diff
@@ -0,0 +1,35 @@
+import { BadRequestException, Injectable, PipeTransform } from '@nestjs/common';
+import AdmZip from 'adm-zip';
+import { LearnroomConfigService } from '../service';
+
+@Injectable()
+export class CommonCartridgeFileValidatorPipe implements PipeTransform<Express.Multer.File, Express.Multer.File> {
+	constructor(private readonly configService: LearnroomConfigService) {}
+
+	public transform(value: Express.Multer.File): Express.Multer.File {
+		if (!value) {
+			throw new BadRequestException('No file uploaded');
+		}
+		if (value.size > this.configService.commonCartridgeImportMaxFileSize) {
+			throw new BadRequestException('File too big');
+		}
+
+		if (!this.hasManifestFile(value.buffer)) {
+			throw new BadRequestException('Invalid file type');
+		}
+
+		return value;
+	}
+
+	private hasManifestFile(file: Buffer): boolean {
+		try {
+			const archive = new AdmZip(file);
+			// imsmanifest.xml is the standard name, but manifest.xml is also valid until v1.3
+			const manifest = archive.getEntry('imsmanifest.xml') || archive.getEntry('manifest.xml');
+
+			return manifest !== null;
+		} catch (error) {
+			return false;

```

Comment:
In the end this is invalid code. How you want to detect if you have a js error, or maybe the AdmZip throw an error. All of this are catched and after it you throw a complet different error, with a message that maybe not releated to the source of the problem. 
When you create a hasXXX Method than they should return a boolean, but do not add try catch arround the body. The error must throw up and must interpret at the executing place of the hasXXX.

Reply:
I renamed the method, to make the intend more clear and added a test. But the code does exactly what it is intended to do. When AdmZip throws -> not a valid Common Cartridge file or no manifest file was found -> not a valid common cartridge file.

Reply:
But you hide the information that maybe the package get wrong input, js code issue or any else. Catching error to false is nearly every time bad practice.